src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -137,29 +114,28 @@ private ProjectInfo CreateCsxProject(string csxPath, IScriptServicesBuilder base
                 return Context.CsxFileProjects[csxPath];
             }
 
-            // Process the file with ScriptCS first
             Logger.LogInformation($"Processing script {csxPath}...");
             Context.CsxFilesBeingProcessed.Add(csxPath);
-            var localScriptServices = baseBuilder.ScriptName(csxPath).Build();
-            var processResult = localScriptServices.FilePreProcessor.ProcessFile(csxPath);
+
+            var fileParser = new FileParser(Context.RootPath);
```

Comment:
to be honest I would love to, and I tried but couldn't find the right APIs. In fact - unless I am mistaken - the necessary APIs ([this?](https://github.com/dotnet/roslyn/blob/0e67586920f6dc2f27474d0d7201134909420c51/src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs#L13)) are not available publicly. What we need here is to walk the dependency tree of `#load` directives and gather all the files and their using statements and so on so that we can construct `ProjectInfo` correctly. It seems the only way to publicly get there with Roslyn scripting APIs at the moment is to manually create a script compilation which I wanted to avoid